crates/typos-cli/src/bin/typos-cli/main.rs

```diff
@@ -156,8 +159,24 @@ fn run_checks(args: &args::Args) -> proc_exit::ExitResult {
 
     let mut typos_found = false;
     let mut errors_found = false;
-    for path in args.path.iter() {
-        let cwd = if path == std::path::Path::new("-") {
+
+    let stdin_paths = if args.stdin_paths {
+        Some(
+            io::stdin()
+                .lines()
+                .map(|res| res.map(PathBuf::from))
+                .collect::<Result<_, _>>()
+                .with_code(proc_exit::sysexits::IO_ERR)?,
+        )
+    } else if args.stdin_paths0 {
+        Some(unix_read_paths_from_stdin()?)
+    } else {
+        None
+    };
+
+    for path in stdin_paths.as_ref().unwrap_or(&args.path) {
+        // Note paths are passed through stdin, `-` is treated like a normal path
+        let cwd = if !args.is_using_stdin_paths() && path == std::path::Path::new("-") {
```

Comment:
Deeper in, we'll still treat `-` as `stdin`, so this specialization right here doesn't quite make sense

Reply:
Really? That doesn't seem ideal. What if someone wanted to lint a codebase that happens to have `-` somewhere?

Reply:
If that is a concern, then solving it is independent of this PR anyways and we should start with an issue from soneone running into this in the wild.

Reply:
Reverted, how it's OK.

Reply:
(BUT :D . I just have to complain about it: these magic strings like `-` are a terrible idea and a bane of CLI software. They were excusable 20 years ago, when they had 1M of RAM and had to write software in C. There's no good reason to do it this way in Rust, where it could be `--stdin` or something like that. No disrespect. Just a pet peeve.)